recipes/c-blosc2/all/conanfile.py

```diff
@@ -0,0 +1,119 @@
+from conans import ConanFile, CMake, tools
+from conan.tools.microsoft import is_msvc
+import os
+import functools
+
+required_conan_version = ">=1.45.0"
+
+class CBlosc2Conan(ConanFile):
+    name = "c-blosc2"
+    description = "A fast, compressed, persistent binary data store library for C."
+    license = "BSD-3-Clause"
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://github.com/Blosc/c-blosc2"
+    topics = ("c-blosc", "blosc", "compression")
+    settings = "os", "arch", "compiler", "build_type"
+    options = {
+        "shared": [True, False],
+        "fPIC": [True, False],
+        "simd_intrinsics": [None, "sse2", "avx2"],
+        "with_lz4": [True, False],
+        "with_zlib": [None, "zlib", "zlib-ng"],
+        "with_zstd": [True, False],
+        "with_plugins": [True, False],
+    }
+    default_options = {
+        "shared": False,
+        "fPIC": True,
+        "simd_intrinsics": "avx2",
+        "with_lz4": True,
+        "with_zlib": "zlib",
+        "with_zstd": True,
+        "with_plugins": True,
+    }
+    generators = "cmake", "cmake_find_package"
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    @property
+    def _build_subfolder(self):
+        return "build_subfolder"
+
+    def export_sources(self):
+        self.copy("CMakeLists.txt")
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+        if self.settings.arch not in ["x86", "x86_64"]:
+            del self.options.simd_intrinsics
+
+    def configure(self):
+        if self.options.shared:
+            del self.options.fPIC
+        del self.settings.compiler.cppstd
+        del self.settings.compiler.libcxx
+
+    def requirements(self):
+        if self.options.with_lz4:
+            self.requires("lz4/1.9.3")
+        if self.options.with_zlib == "zlib-ng":
+            self.requires("zlib-ng/2.0.6")
+        elif self.options.with_zlib == "zlib":
+            self.requires("zlib/1.2.12")
```

Comment:
With https://github.com/Blosc/c-blosc2/blob/v2.2.0/CMakeLists.txt#L153-L164, how do you ensure that zlib is used rather than zlib-ng?

Reply:
Thank you for this as well!
I will check this along with the cmake module issue above.

Reply:
When with_zlib is zlib, I think zlib is used because zlib-ng is not installed.
Is it wrong?

Reply:
Probably it is easier to patch that file (or use `replace_in_file`) and use something more straightforward, wdyt?

Reply:
@SpaceIm  @jgsogo 
I update patch file to `find_package` zlib  only, because c-blosc2 uses zlib-ng with compat mode.
https://github.com/Blosc/c-blosc2/blob/v2.2.0/CMakeLists.txt#L171

zlib-ng with compat mode provides "zlib" package.